### DISCUSSION TITLE
Is transcluding of tiddlers in popup windows planned?

### DISCUSSION THREAD

**Discussion by teampolyglot** [crazko/krystal#36] [Ideas]
I love this theme! Thank you for making it. :heart_eyes: I'm going to include this by default to all users of https://learnawesome.org

Any chance that showing tiddler preview on hover will be supported in future? Here's an [example](https://giffmex.org/gifts/transclusioninpopups.html) of it.

**Comment by crazko** [comment]
Thanks, I'm glad you like it!

I was thinking about making such feature, but then I found this plugin https://tobibeer.github.io/tw5-plugins/#preview. Haven't tried it yet, but it looks working well. Maybe it would be enough for your use case?

**Comment by fkohrt** [comment]
The [preview plugin](https://github.com/tobibeer/tw5-preview) would be a perfect fit indeed, but it doesn't work well together with Krystal right now. The popups often appear outside of the screen.

Which also made me notice #5: Installing @tobibeer's plugins required me to first turn off Krystal, then install them and afterwards re-enable Krystal.

**Comment by crazko** [comment]
I see, I'll have to find time and test the integration of this plugin. It's better to use existing one than to develop it from the scratch :)

**Comment by crazko** [comment]
I checked it and unfortunately those two aren't compatible, due to how positioning works in the preview plugin. Also I do not see a way for the positioning configuration.

[Tippy.js](https://atomiks.github.io/tippyjs/) looks like a perfect candidate for the tooltip logic, but I didn't originally planned this feature. Let's see if there're more people who lack it...

**Comment by keelancook** [comment]
This may be related concerning the Refnotes plugin. At current, with the footnote macro, if the footnote lands near the right edge of the text on a tiddler, the popup opens to the right of the note and forces the width of the tiddler text body to be wider than the actual frame for the tiddler. This results in "wiggle" inside the tiddler from left to right. 

Personally, I don't know enough about the how plugins play with one another to know if this is an issue internal to Refnotes (they have it coded to always open the popup to the right of the number, even when it hits the edge of a tiddler), or if it is something to do with the way Krystal reveals a popup. 

This is a small annoyance that I'm fine overlooking, but I thought I would report it here in what appeared to be a related discussion. Happy to provide further explanation if it is helpful for you.

**Comment by crazko** [comment]
I probably understand, what you mean: 
![image](https://user-images.githubusercontent.com/1255989/86962988-0e5ba980-c164-11ea-8811-356dc82841c5.png)

I'm afraid I'm not able to change Krystal in a way to make it compatible with Refnotes footnote macro. It would need to change how the popup is generated in the HTML markup.

---

So far there are two plugins ([Refnotes](https://kookma.github.io/Refnotes/), [Preview](https://github.com/tobibeer/tw5-preview)) providing tiddler transclusion in popups, but neither one is compatible with this plugin due to way of how popups are generated.

**Comment by keelancook** [comment]
That was my guess. I assumed if it could be changed at all, that it would be in the other plugin. Is the HTML markup for those popups something that is designated by the other plugin? Clearly, I know very little about the whereabouts of those things, haha.

Thanks again.

**Comment by crazko** [comment]
> Is the HTML markup for those popups something that is designated by the other plugin?

Yes, it is, so I can't do anything with that. Maybe it would worth a shot to ask the authors of those plugins whether they would  consider a change in the behavior.

**Comment by keelancook** [comment]
I may check in with them and alert them to the behavior.

Thanks!

**Comment by garrettvercoe** [comment]
I also tried out Giffmex's solution, https://giffmex.org/gifts/transclusioninpopups.html. 
Still has the cutoff due to the position:sticky of the Tiddler frame. 

I think this could be a [possible workaround](https://css-tricks.com/popping-hidden-overflow/), but I have trouble figuring out how to translate it to TiddlyWiki

**Comment by kookma** [comment]
> I'm afraid I'm not able to change Krystal in a way to make it compatible with Refnotes footnote macro. It would need to change how the popup is generated in the HTML markup.

The issue is with Tiddlywiki itself! It has no mechanism to reposition the popup and tooltip specially on mobile screen when they go off screen! There are some discussion in Forum on this and Refnotes still suffers this issue! The best solution by now is Repopup see second ref below

Ref:
- https://groups.google.com/g/tiddlywiki/c/dFxN3eIrAzo/m/QZP0vp1uDAAJ
- https://thesherwood.github.io/Mentat/  (Important: See Repopup)
